src/cloudinary/helper.js

```diff
@@ -0,0 +1,93 @@
+import { A11Y_TRANSFORMS, progressive, PLACEHOLDER_OPTIONS, COMPONENTS, predominantColorTransformPxl } from '../constants';
+
+/**
+ * 
+ * @param {Boolean} enable 
+ * @returns {Object} transformation object for progressive effect
+ */
+const addProgressive = (enable = false) => {
+  return enable ? progressive : {}
+}
+
+/**
+ * 
+ * @param {'darkmode'|'brightmode'|'monochrome'|'colorblind'} type 
+ * @returns {Object} transformation effect for a11y support based on type
+ */
+const addA11y = (type) => {
+  return A11Y_TRANSFORMS[type] || {}
+}
+
+/**
+ * 
+ * @param {Object} object contains accessibility, withProgressive, cldTransforms and baseOptions
+ * @returns {Object} options for generating delivery URL of a media component 
+ */
+export const computeOptions = ({ accessibility, withProgressive, baseOptions = {}, extra = [] }) => {
+  const transformation = [...extra]
+  const a11y = addA11y(accessibility)
+  const progressive = addProgressive(withProgressive)
+
+  transformation.push(progressive)
+  transformation.push(a11y)
+
+  return {
+    ...baseOptions,
+    transformation,
+  }
+}
+
+/**
+ * 
+ * @param {Object} baseOptions 
+ * @param {Object} extra 
+ * @returns {Object} new options Object with extra inside transformation
+ */
+export const extendOptions = (baseOptions = {}, extra) => {
+  let transformation = baseOptions.transformation ? [...baseOptions.transformation] : []
+
+  if (Array.isArray(extra)) {
+    extra.forEach(effect => transformation.push(effect))
+  } else if (extra) {
+    transformation.push(extra)
+  }
+
+  return {
+    ...baseOptions,
+    transformation
+  }
+}
+
+/**
+ * 
+ * @param {'lqip'|'color'|'pixelate'|'predominant-color'|'vectorize'|'blur'} type 
+ * @param {Option} imgOptions 
```

Comment:
This "Option" type is not declared, did you mean Object?

Reply:
Object :D